### PR TITLE
Feature/public profile

### PR DESCRIPTION
**Pull request by vic2019**
- [x] A user's public profile with a sharable url
- [x] 'Friends' tab showing people that user follows and suggested people (right now just all the ppl in db)
- [x] Button to follow and unfollow another user

1) All the public profiles in the front end are stored in `discover` as an object with userId's as keys

2) The current controllers prevent a user from fetching lists owned by other people. I created a lists endpoint at a different URL (/public/:userId) for that purpose. But we could also make the original endpoint public and use it for both fetching one's own lists and other people's lists. What to do?

3) I added a PUT handler for creating/deleting follows. I ran into a lot of trouble when a second DELETE would cause an error, even though the resource to be deleted was already gone. Using the DELETE method is fine too, though we might have to make it return 200 for cases where the resource to be deleted cannot be found.

### CODE REVIEW

**Review by aeciorc — COMMENTED** [review]
That looks really good. From a UX perspective, I'd change two things:
- When I click "follow" or "following", the user disappears from the tab. It'd be better if it stays there and the button changed instead (allowing me to undo the operation)
- When I scroll down, the tabs should stay at the top (sticky positioning) so that I don't have to scroll back up to switch between tabs

**Comment by maismin** [comment]
I think that `friends` should be changed to `follows` as you could be following one user, but the user might not necessarily be following you.

**Comment by maismin** [comment]
> 2. The current controllers prevent a user from fetching lists owned by other people. I created a lists endpoint at a different URL (/public/:userId) for that purpose. But we could also make the original endpoint public and use it for both fetching one's own lists and other people's lists. What to do?

Since you've already created a public route for public pages, we could leave it like that. I could see the use case for having both a public and protected route for the user if later down the line we decide to add more features that requires both public and protected routes.

> 3. I added a PUT handler for creating/deleting follows. I ran into a lot of trouble when a second DELETE would cause an error, even though the resource to be deleted was already gone. Using the DELETE method is fine too, though we might have to make it return 200 for cases where the resource to be deleted cannot be found.

A 404 should be returned if a user attempts to unfollow something that doesn't exist.

File: client/src/pages/Discover.js [hunk 1]
```diff
@@ -0,0 +1,16 @@
+/* eslint-disable no-unused-vars */
+import React from 'react';
+import { Toolbar } from '@material-ui/core';
+
+import DiscoveredUsersList from '../components/Discover/DiscoveredUsersList';
```
Comment (maismin) [inline client/src/pages/Discover.js]: The `/Discover` directory isn't in the PR.

Comment (vic2019) [reply client/src/pages/Discover.js]: I removed that import

**Comment by maismin** [comment]
The User shouldn't be able to follow his own profile.

**Review by Esodey — APPROVED** [review]
All looks great. Smoothly integrated into our existing set up. Well done

**Comment by vic2019** [comment]
> The User shouldn't be able to follow his own profile.

I disabled the FollowButton for the user's own profile. The user would still show up under Suggested (without the ability to follow itself) because the suggested feature right now just returns all the users in the db

**Pull request merged**